plugins/microservices/src/get_server_property.cpp

```diff
@@ -0,0 +1,105 @@
+            return SYS_INTERNAL_NULL_INPUT_ERR;
+        }
+
+        // ensure that this user is an admin
+        if (rei->uoic->authInfo.authFlag < LOCAL_PRIV_USER_AUTH) {
+            log_msi::error("msi_get_server_property: User {} is not local admin. status = {}",
+                           rei->uoic->userName,
+                           CAT_INSUFFICIENT_PRIVILEGE_LEVEL);
+            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
+        }
+
+        try {
+            std::string name = parseMspForStr(_in_name);
+            const nlohmann::json* json_value = nullptr;
+            const auto& config = irods::server_properties::server_properties::instance().map();
+
+            try {
+                if (name.find("/") != std::string::npos) {
+                    json_value = &config.at(nlohmann::json::json_pointer(name));
+                }
+                else {
+                    json_value = &config.at(name);
+                }
+            }
+            catch (nlohmann::json::out_of_range& e) {
+                log_msi::error("Property not found. An error occurred {}", e.what());
+                return KEY_NOT_FOUND;
+            }
+            catch (nlohmann::json::parse_error& e) {
+                log_msi::error("Parse error in key. {}", e.what());
+                return KEY_NOT_FOUND;
+            }
+
+            const std::string value = json_value->dump();
+            fillStrInMsParam(_out_value, value.c_str());
+            return 0;
+        }
+        catch (const irods::exception& e) {
+            log_msi::error(e.what());
+            return SYS_INTERNAL_ERR;
+        }
+        catch (const std::exception& e) {
+            log_msi::error(e.what());
+            return SYS_INTERNAL_ERR;
+        }
+        catch (...) {
+            log_msi::error("An unknown error occurred while processing the request.");
+            return SYS_UNKNOWN_ERROR;
+        }
+    } // msi_impl
+
+    template <typename... Args, typename Function>
+    auto make_msi(const std::string& name, Function func) -> irods::ms_table_entry*
+    {
+        auto* msi = new irods::ms_table_entry{sizeof...(Args)};
+        msi->add_operation(name, std::function<int(Args..., ruleExecInfo_t*)>(func));
+        return msi;
+    } // make_msi
+} // anonymous namespace
+
+extern "C" auto plugin_factory() -> irods::ms_table_entry*
+{
+    return make_msi<msParam_t*, msParam_t*>("msi_get_server_property", msi_impl);
+}
+
+#ifdef IRODS_FOR_DOXYGEN
+/// \brief Gets the value of a value in the server_properties map. Requires administrative
+/// permissions
+///
+/// \param[in]      in_name   The name of the property to retrieve, may be a json pointer or a simple
+/// name
+/// \param[out]     out_value The value of the property. It is in serialized form if it is an object,
+/// otherwise it is converted to a string
+/// \param[in,out]  rei       The rule execution context. Ignore this parameter as a user.
```

Comment:
Format the comments so that they are easier to read in code.
```cpp
/// \param[in]     in_name   The name of the property to retrieve, may be a json pointer or a simple
///                          name.
/// \param[out]    out_value The value of the property. It is in serialized form if it is an object,
///                          otherwise it is converted to a string.
/// \param[in,out] rei       The rule execution context. Ignore this parameter as a user.
```